docs/samples/Microsoft.ML.Samples/Dynamic/Transforms/ImageAnalytics/ConvertToGrayScale.cs

```diff
@@ -42,29 +43,41 @@ public static void Example()
                            .Append(mlContext.Transforms.ConvertToGrayscale("Grayscale", "ImageObject"));
 
             var transformedData = pipeline.Fit(data).Transform(data);
-            // The transformedData IDataView contains the loaded images column, and the grayscaled column.
 
-            // Preview the transformedData. 
-            var transformedDataPreview = transformedData.Preview();
-            PrintPreview(transformedDataPreview);
-            // ImagePath    Name         ImageObject            Grayscale
-            // tomato.bmp   tomato       System.Drawing.Bitmap  System.Drawing.Bitmap
-            // banana.jpg   banana       System.Drawing.Bitmap  System.Drawing.Bitmap
-            // hotdog.jpg   hotdog       System.Drawing.Bitmap  System.Drawing.Bitmap
-            // tomato.jpg   tomato       System.Drawing.Bitmap  System.Drawing.Bitmap
+            PrintColumns(transformedData);
+            // ImagePath    Name         ImageObject              Grayscale
+            // tomato.bmp   tomato       {Width=800, Height=534}  {Width=800, Height=534}
+            // banana.jpg   banana       {Width=800, Height=288}  {Width=800, Height=288}
+            // hotdog.jpg   hotdog       {Width=800, Height=391}  {Width=800, Height=391}
+            // tomato.jpg   tomato       {Width=800, Height=534}  {Width=800, Height=534}
         }
 
-        private static void PrintPreview(DataDebuggerPreview data)
+        private static void PrintColumns(IDataView transformedData)
         {
-            foreach (var colInfo in data.ColumnView)
-                Console.Write("{0,-25}", colInfo.Column.Name);
-
-            Console.WriteLine();
-            foreach (var row in data.RowView)
+            Console.WriteLine("{0, -25} {1, -25} {2, -25} {3, -25}", "ImagePath", "Name", "ImageObject", "Grayscale");
+            using (var cursor = transformedData.GetRowCursor(transformedData.Schema))
             {
-                foreach (var kvPair in row.Values)
-                    Console.Write("{0,-25}", kvPair.Value);
-                Console.WriteLine();
+                var imagePathGetter = cursor.GetGetter<ReadOnlyMemory<char>>(cursor.Schema["ImagePath"]);
+                var nameGetter = cursor.GetGetter<ReadOnlyMemory<char>>(cursor.Schema["Name"]);
+                var imageObjectGetter = cursor.GetGetter<Bitmap>(cursor.Schema["ImageObject"]);
+                var grayscaleGetter = cursor.GetGetter<Bitmap>(cursor.Schema["Grayscale"]);
+                while (cursor.MoveNext())
+                {
+                    ReadOnlyMemory<char> imagePath = null;
+                    imagePathGetter(ref imagePath);
```

Comment:
While you have the getter outside the while loop, which is good, I see we have the variable inside the while loop. This we generally try to avoid as it is a bad practice. We try to do this throughout our codebase and implementations, and we should continue to encourage this "habit" in our example code, since it is something for people to learn from.

As it happens a few days ago I wrote a sample where I made this advice explicit.

[https://github.com/dotnet/machinelearning/blob/66ff419a95ae92dc80478b5e4af21002aae9ded0/docs/samples/Microsoft.ML.Samples/Dynamic/SimpleDataViewImplementation.cs#L40-L49](https://github.com/dotnet/machinelearning/blob/66ff419a95ae92dc80478b5e4af21002aae9ded0/docs/samples/Microsoft.ML.Samples/Dynamic/SimpleDataViewImplementation.cs#L40-L49)

Reply:
Ok, great. thanks will update.